activity/journal.go

```diff
@@ -0,0 +1,67 @@
+package activity
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+// Journal contains the command-line used to invoke the process associated with this item
+// and its journal ID.
+type Journal struct {
+	ID, Description string
+	Start           time.Time
+}
+
+var mux = sync.Mutex{}
+
+func initHistory() ([]Journal, map[string]struct{}) {
+	return make([]Journal, 0), make(map[string]struct{})
+}
+
+// history: a list of commands that have been invoked
+// jidToHistory: lookup to identify whether a journal ID has already been recorded in history.
+var history, jidToHistory = initHistory()
+
+// registerCommand appends the command description to history if it hasn't been registered before.
+func (j Journal) registerCommand() {
```

Comment:
`registerCommand` => `addToHistory` -- the latter better captures the generic nature of these journals since they do not have to be limited to just commands.